Polyfills/XMLHttpRequest/Source/XMLHttpRequest.cpp

```diff
@@ -126,60 +162,35 @@ namespace Babylon
         }
     }
 
-    void XMLHttpRequest::Open(const Napi::CallbackInfo& info)
+    void XMLHttpRequest::Abort(const Napi::CallbackInfo&)
     {
-        m_method = info[0].As<Napi::String>().Utf8Value();
-        m_url = GetAbsoluteUrl(info[1].As<Napi::String>().Utf8Value(), m_rootUrl);
-        SetReadyState(ReadyState::Opened);
+        m_request.Abort();
     }
 
-    void XMLHttpRequest::Send(const Napi::CallbackInfo& info)
+    void XMLHttpRequest::Open(const Napi::CallbackInfo& info)
     {
-        // TODO: Handle exceptions
-        SendAsync();
+        m_request.Open(MethodType::StringToEnum(info[0].As<Napi::String>().Utf8Value()), info[1].As<Napi::String>().Utf8Value());
+        SetReadyState(ReadyState::Opened);
     }
 
-    // TODO: Make this just be SendAsync() once the UWP file access bug is fixed.
-    arcana::task<void, std::exception_ptr> XMLHttpRequest::SendAsyncImpl()
+    void XMLHttpRequest::Send(const Napi::CallbackInfo& info)
     {
-        if (m_responseType.empty() || m_responseType == XMLHttpRequestTypes::ResponseType::Text)
-        {
-            return LoadTextAsync(m_url).then(m_runtimeScheduler, arcana::cancellation::none(), [this](std::string data) {
-                // check UTF-8 BOM encoding
-                if (data.size() >= 3 && data[0] == '\xEF' && data[1] == '\xBB' && data[2] == '\xBF')
-                {
-                    m_responseText = data.substr(3);
-                }
-                else
+        m_request.SendAsync()
+            .then(m_runtimeScheduler, arcana::cancellation::none(), [this](arcana::expected<void, std::exception_ptr> result) {
+                if (result.has_error())
                 {
-                    // UTF8 encoding
-                    m_responseText = std::move(data);
+                    throw result.error();
                 }
```

Comment:
Unless I'm missing something, taking an `expected`, checking for an error, and just throwing it is equivalent to just passing a callable to `then` that takes just the result of the antecedent task (in this case `void`).

Reply:
Oh yeah, good point. I was trying to fail on this, but that isn't actually going to work in a continuation.